gno/p/basedao/basedao.gno

```diff
@@ -0,0 +1,118 @@
+package basedao
+
+import (
+	"errors"
+	"std"
+
+	"gno.land/p/demo/mux"
+	"gno.land/p/teritori/daocond"
+	"gno.land/p/teritori/daokit"
+)
+
+type DAO struct {
+	Core             *daokit.Core
+	Members          *MembersStore
+	RenderRouter     *mux.Router
+	GetProfileString ProfileStringGetter
+	Realm            std.Realm
+}
+
+type Config struct {
+	Name              string
+	Description       string
+	ImageURI          string
+	Members           *MembersStore
+	NoDefaultHandlers bool
+	InitialCondition  daocond.Condition
+	SetProfileString  ProfileStringSetter
+	GetProfileString  ProfileStringGetter
+}
+
+type ProfileStringSetter func(field string, value string) bool
+type ProfileStringGetter func(addr std.Address, field string, def string) string
+
+func New(conf *Config) *DAO {
+	members := conf.Members
+	if members == nil {
+		members = NewMembersStore(nil, nil)
+	}
+
+	if conf.GetProfileString == nil {
+		panic(errors.New("GetProfileString is required"))
+	}
+
+	dao := &DAO{
+		Core:             daokit.NewCore(),
+		Members:          members,
+		GetProfileString: conf.GetProfileString,
+		Realm:            std.CurrentRealm(),
+	}
+	dao.initRenderingRouter()
+
+	if conf.SetProfileString != nil {
+		conf.SetProfileString("DisplayName", conf.Name)
+		conf.SetProfileString("Bio", conf.Description)
+		conf.SetProfileString("Avatar", conf.ImageURI)
+	}
+
+	if !conf.NoDefaultHandlers {
+		if conf.SetProfileString == nil {
+			panic(errors.New("SetProfileString is required"))
+		}
+
+		if conf.InitialCondition == nil {
+			conf.InitialCondition = daocond.MembersThreshold(0.6, members.IsMember, members.MembersCount)
+		}
+
+		if conf.SetProfileString != nil {
+			dao.Core.SetResource(&daokit.Resource{
+				Handler:   NewEditProfileHandler(conf.SetProfileString, []string{"DisplayName", "Bio", "Avatar"}),
+				Condition: conf.InitialCondition,
+			})
+		}
+
+		// register management handlers
+		for _, m := range []daokit.MessageHandler{
```

Comment:
shouldn't this be dao_interfaces.MessageHandler ?
```suggestion
		for _, m := range []dao_interfaces.MessageHandler{
```

Reply:
dao_interfaces is the old framework, it's still used by social feed and front (thus e2e tests) so I kept it